pyquil/paulis.py

```diff
@@ -282,6 +308,12 @@ def __radd__(self, other):
         assert isinstance(other, Number)
         return self + other
 
+    def __sub__(self, other):
+        return self + -1. * other
+
+    def __rsub__(self, other):
+        return other + -1. * self
+
```

Comment:
docstrings were added on the PauliTerm built in methods because they have some unexpected behaviour, e.g. returning PauliSums not Pauliterms